akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphMergeSortedSpec.scala

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.scaladsl
+
+import akka.stream._
+import akka.stream.scaladsl._
+import akka.stream.testkit.TwoStreamsSetup
+import org.scalacheck.Gen
+import scala.util.Random
+import org.scalatest.prop.GeneratorDrivenPropertyChecks
+import org.scalatest.concurrent.ScalaFutures
+import scala.concurrent.duration._
+import org.scalactic.ConversionCheckedTripleEquals
+import org.scalacheck.Shrink
+
+class GraphMergeSortedSpec extends TwoStreamsSetup with GeneratorDrivenPropertyChecks with ScalaFutures with ConversionCheckedTripleEquals {
+  import GraphDSL.Implicits._
+
+  override type Outputs = Int
+
+  override def fixture(b: GraphDSL.Builder[_]): Fixture = new Fixture(b) {
+    val merge = b.add(new MergeSorted[Outputs])
+
+    override def left: Inlet[Outputs] = merge.in0
+    override def right: Inlet[Outputs] = merge.in1
+    override def out: Outlet[Outputs] = merge.out
+  }
+
+  implicit val patience = PatienceConfig(1.second)
+  implicit def noShrink[T] = Shrink[T](_ ⇒ Stream.empty) // do not shrink failures, it only destroys evidence
+
+  "MergeSorted" must {
+
+    "work in the nominal case" in {
+      val gen = Gen.listOf(Gen.oneOf(false, true))
```

Comment:
"pulling in" as in using it here. In more complex cases I almost agree with you, but  Gen.listOf(Gen.oneOf(false, true)) does not seem to do much here. My other problem is with ScalaCheck is that it can gives false confidence (see the sliding implementation). I would prefer to see the list of cases being checked than a black-box. Especially since these corner-cases allow in a possible rewrite to analyze what issues are to be faced.